app/main.py

```
@@ -1,25 +1,79 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: [ABC], name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: ABC, owner: [ABC]) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: ABC, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
```

In other case you should raise an error